src/bin/ipa_bench/gen_events.rs

```diff
@@ -16,35 +16,96 @@ use std::time::Duration;
 const RECORD_SEPARATOR: u8 = 30;
 
 const DAYS_IN_EPOCH: u64 = 7;
-type MatchKey = Vec<u64>;
+type MatchKey = u64;
 type Epoch = u8;
+type Number = u32;
 
 #[derive(Clone)]
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 pub struct EventBase {
-    // For this tool, we'll fix the length of a matchkey to u64
-    pub matchkeys: MatchKey,
+    /// Secret shared and then encrypted match keys.
+    pub matchkeys: Vec<MatchKey>,
+
+    /// The epoch which this event is generated. Using an 8-bit value = 256 epochs > 4 years (assuming 1 epoch = 1 week).
+    /// This field is in the clear.
     pub epoch: Epoch,
-    pub timestamp: u32,
+
+    /// An offset in seconds into a given epoch. The clear is u32 (< 2^20 seconds), then encrypted and secret shared.
+    pub timestamp: Number,
+
+    /// 0 (false) if this is a source event. 1 (true) otherwise.
+    pub flag: bool,
 }
 
 #[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
 pub struct SourceEvent {
     pub event: EventBase,
 
     /// Ad breakdown key value
-    pub breakdown_key: String,
+    pub breakdown_key: Number,
+
+    /// Fields from TriggerEvent for padding
+    value: Number,
+    zkp: Number,
```

Comment:
I suggest we remove this. It's not in our document, and we may not even need it. The "capping" thing should already protect pretty well against this type of attack.